### DISCUSSION TITLE
[Feauture Request] Add helper function for composing class mixins

### DISCUSSION THREAD

**Discussion by mihaisavezi** [ing-bank/lion#26] [General]
This helps when there several mixins applied to a class, i.e. in ```LionFieldset.js```

```javascript

// transform this 
export class LionFieldset extends FormControlMixin(
  ValidateMixin(EventMixin(CssClassMixin(SlotMixin(ObserverMixin(LionLitElement))))),
) {...}

// to this
// and mixins are applied in order, from left to right
export class LionFieldset extends pipe(
  ObserverMixin,
  SlotMixin,
  CssClassMixin,
  EventMixin,
  ValidateMixin,
  FormControlMixin,
)(LionLitElement) {
```

Several other helper libraries already expose this functionality, lodash for [example](https://github.com/lodash/lodash/wiki/FP-Guide). It's basically just a helper for function composition.

**Comment by LarsDenBakker** [comment]
I propose using mixin syntax, which it easier to understand and read what's going on. And is similar in syntax of other languages, such as Scala:

```js
export class LionFieldset extends mix(LionLitElement).with(
  ObserverMixin,
  SlotMixin,
  CssClassMixin,
  EventMixin,
  ValidateMixin,
  FormControlMixin,
) {
```

An example library is https://github.com/justinfagnani/mixwith.js though it's small enough to implement it custom.

It also mimicks a proposal for mixin syntax: https://github.com/justinfagnani/proposal-mixins

**Comment by bashmish** [comment]
IMO you can just use Justin's [mixwith](https://www.npmjs.com/package/mixwith) directly. Why do you want to have it in `@lion`?

I'm thinking towards closing this as a wontfix, do you have any considerations? cc @daKmoR @tlouisse

**Comment by erikkroes** [comment]
Hi,

We are closing this issue because we're changing the way we handle feature requests. Your issue will get the "votes needed" label, and people can add 👍 reactions to [vote and show interest](https://github.com/ing-bank/lion/issues?q=label%3A%22votes+needed%22+sort%3Areactions-%2B1-desc). It will be reopened when this feature gets picked up. 

This we way we clearly focus on bugs in our open issues. At the same time we can see what community interest there is for new features.

Thanks!
